### ISSUE TITLE
Enhancement: [ban-types] Allow {} in the recommended config

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8697]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/ban-types

### Description

The `{}` has long been a source of confusion in TypeScript. It means _"any non-nullish value, but with no known fields"_. That includes objects like `{}` and `{ value: 1 }`, primitives like `true` and `""`, arrays like `[]`, and general class instances like `new Error()`.

Because users so often get confused by `{}` allowing primitives like `true` and `""`, the `ban-types` rule suggests users use something more restrictive. Specifically its default options include:

> ```plaintext
> `{}` actually means "any non-nullish value".',
> - If you want a type meaning "any object", you probably want `object` instead.',
> - If you want a type meaning "any value", you probably want `unknown` instead.',
> - If you want a type meaning "empty object", you probably want `Record<string, never>` instead.',
> - If you really want a type meaning "any non-nullish value", you probably want `NonNullable<unknown>` instead.',
> ```

That's pretty strict... at the cost of user convenience.

Now that #8364 is in, we have the ability to provide more lenient default options for the `recommended*` configs vs. the `strict*` configs.

Proposal: let's remove the restriction on `{}` from the default `ban-types` options in `recommended` configs, and leave them in for `strict` configs?

### Fail

```TypeScript
let value: Object;
```

### Pass

```TypeScript
let value: {};
```

### Additional Info

@RyanCavanaugh, the dev lead for TypeScript, is not a fan of the way it is now 😄: https://bsky.app/profile/searyanc.dev/post/3knqtqcjhpn2d 

See also #5947 for past discussion around objects and `ban-types`

**Comment by JoshuaKGoldberg** [comment]
Just read https://github.com/microsoft/TypeScript/issues/57735 and filed #8700 as a result. For clarity, the intent of the two issues I'm filing are:
* #8700: holistically rethinking `ban-types`, which might take some time
* This one: spot improving the `recommended` experience, which can land first

If #8700 is resolved before this one, that's great too.

**Comment by bradzacher** [comment]
Strong 👎
See https://github.com/typescript-eslint/typescript-eslint/issues/8700#issuecomment-2002610682

**Comment by Josh-Cena** [comment]
From what I understand: this is about removing the ban in `recommended`, while #8700 is about removing the ban everywhere...

**Comment by bradzacher** [comment]
They're one and the same though. The rule supports banning `{}` but it only flags it because it's in the default config. If you remove it from the default config then it's removed from everywhere.

**Comment by JoshuaKGoldberg** [comment]
As of #8364 we can have different default options for `recommended` and `strict`. That's what I meant for this issue.

But yeah, I wasn't sure whether to file this and/or #8700 as a single issue, or discussion, or etc.. Marking this one as blocked on resolving discussion there in the interim.

**Comment by JoshuaKGoldberg** [comment]
Closing per https://github.com/typescript-eslint/typescript-eslint/issues/8700#issuecomment-2073061412. We can always revisit this if v9 shows that many users are still confused by the `{}` ban with the new, better-phrased rule.
